dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/ManagedBeanOBase.java

```diff
@@ -147,10 +147,18 @@ protected void injectInstance(ManagedObject<?> managedObject, Object instance, I
                     }
                 }
             } catch (Throwable t) {
-                FFDCFilter.processException(t, CLASS_NAME + ".injectInstance", "134", this);
-                if (isTraceOn && tc.isEntryEnabled())
-                    Tr.exit(tc, "injectInstance : Injection failure", t);
-                throw ExceptionUtil.EJBException("Injection failure", t);
+                if (! (t instanceof ManagedObjectException)) {
+                    FFDCFilter.processException(t, CLASS_NAME + ".injectInstance", "134", this);
+                    if (isTraceOn && tc.isEntryEnabled())
+                        Tr.exit(tc, "injectInstance : Injection failure", t);
+                    throw ExceptionUtil.EJBException("Injection failure", t);
+                } else {
+                    Throwable cause = t.getCause();
+                    FFDCFilter.processException(cause, CLASS_NAME + ".injectInstance", "134", this);
```

Comment:
technically, ever FFDC should have a different probe ID (usually line number), I'd change this one to "157"